src/overcooked_ai_py/mdp/layout_generator.py

```diff
@@ -76,29 +78,59 @@ def mdp_generator_fn():
                 inner_shape=[rnd_int_uniform(*dim) for dim in size_bounds],
                 prop_empty=rnd_uniform(*prop_empty),
                 prop_features=rnd_uniform(*prop_feats),
+                feature_types=feature_types,
                 display=display
             )
 
         return mdp_generator_fn
 
+    @staticmethod
+    def add_generated_mdp_params_orders(mdp_params):
+        """
+        adds generated parameters (i.e. generated orders) to mdp_params
+        """
+        mdp_params = copy.deepcopy(mdp_params)
+        if mdp_params.get("generate_all_orders"):
+            all_orders_kwargs = copy.deepcopy(mdp_params["generate_all_orders"])
+
+            if all_orders_kwargs.get("recipes"):
+                 all_orders_kwargs["recipes"] = [r if isinstance(r, Recipe) else Recipe.from_dict(r) for r in all_orders_kwargs["recipes"]]
+        
+            all_recipes = Recipe.generate_random_recipes(**all_orders_kwargs)
+            mdp_params["start_all_orders"] = [r.to_dict() for r in all_recipes]
```

Comment:
It's important that `mdp_params` is JSON serializable, as saving and loading agents in the client code relies on this. Thus, you should probably only support recipe dicts as input, since Recipe instances are not JSON serializable.

Reply:
replaced `[r if isinstance(r, Recipe) else Recipe.from_dict(r) for r in all_orders_kwargs["recipes"]]` with `[Recipe.from_dict(r) for r in all_orders_kwargs["recipes"]]`